src/App/routes/user.js

```diff
@@ -0,0 +1,18 @@
+import { Router } from 'express';
+import { getUsers, createUser, deleteUser, getUser, updateUser } from '../controllers/user';
+import { signupValidate } from '../middlewares/validator';
+import { isAdmin }  from '../middlewares/permission';
+import token  from '../middlewares/token';
+import protect from '../middlewares/protect';
+
+const router = Router();
+
+router.route('/')
+    .get(getUsers)
+    .post(token, isAdmin, protect, signupValidate, createUser);
```

Comment:
It doesn't make sense at all to use token and protect which do the same exact or nearly close works!